### DISCUSSION TITLE
useSWR from a redux-thunk?

### DISCUSSION THREAD

**Discussion by llaenowyd** [vercel/swr#885] [Q&A] [answered]
I just need a bit of server data as context while doing this logic, and otherwise that server data is supplied to the application with `useSWR`...

I could make a component that `onEffect` loads the `useSWR` data into the redux store, then it's available to the thunk.

Any better way?

**Comment by koba04** [comment, marked as answer]
What about passing `data` retrieved from `useSWR` to your think function as an argument?

```js
const { data } = useSWR('/api/user', fetcher)
return <button onClick={() => {
  dispatch(yourThunk(data));
}}>click</button>;
```

**Comment by hc0503** [comment]
I implemented similar with bellow code.

```js
const { error } = useSWR("/api/books", dispatch(fetchBook(router.query.id)));
```
And I displayed redux data in component.
